### DISCUSSION TITLE
Lesson 10 challenge

### DISCUSSION THREAD

**Discussion by lamtrinh259** [Cyfrin/foundry-full-course-cu#365] [Q&A]
This is supposed to be an easier challenge, but I'm getting stuck. Any help or hint? I've tried cast send also, but doesn't work...

**Comment by asobiesk** [comment]
Take a look at what has to successfully happen in order for you to call solveChallenge without reverting.

```solidity
bool success = i_erc20.transferFrom(msg.sender, address(this), COST_TO_SOLVE);
```

Looks like some ERC20 token has to be transferred successfully between two accounts. So, what are the criteria for a successful transfer of a token? Surely the account that the token is transferred from has to have sufficient balance of that particular token. Is that enough or was there some additional step required...?

Hope this helps :)

**Reply by lamtrinh259** [reply]
@asobiesk hi, yes, I've seen the code and I'm aware of what I need to "sacrifice" in order to solve the challenge. So I've tried and neither have worked:
1. Sending the "sacrificial thing" directly to lesson 10 contract
2. Approve the spending of the sacrificial thing and then send to the contract manually
3. Solve challenge directly while having the sacrificial thing

Not sure what I may be missing...

**Reply by asobiesk** [reply]
Try combining the first part of option 2 with the option 3 :)

**Reply by sharonphiliplima** [reply]
The question is how does @lamtrinh259  get to approve the sacrificial thing and also mint it beforehand?
Hint: refer back to what @PatrickAlphaC taught about the mint function in the Test file in Lesson 10 : )

**Reply by 0xTagger** [reply]
I've tried what @lamtrinh259 did and then @asobiesk suggested and was not able to solve it. I also went back to the file you suggested in Lesson 10 and am still confused.
What am I missing? Any feedback is appreciated :)

EDIT: Figured it out 👍
